backend/benefit/users/tests/test_gdpr_api.py

```diff
@@ -0,0 +1,46 @@
+import json
+from typing import Optional
+
+import pytest
+from django.contrib.auth import get_user_model
+from django.test import override_settings
+from django.urls import reverse
+
+from shared.common.tests.factories import HelsinkiProfileUserFactory
+from users.models import format_date
+
+User = get_user_model()
+
+
+def extract_value(gdpr_api_response: dict, key: str) -> Optional[str]:
+    """Extract value from GDPR API response."""
+    user: dict = gdpr_api_response["children"]
+    return next((item["value"] for item in user if item["key"] == key), None)
+
+
+@override_settings(NEXT_PUBLIC_MOCK_FLAG=True)
+def test_get_profile_data_from_gdpr_api_no_auth(api_client):
+    """Test that the profile data is valid without authentication."""
+    user = HelsinkiProfileUserFactory()
+    response = api_client.get(reverse("gdpr_v1", kwargs={"uuid": user.username}))
+    response_dict = json.loads(response.content)
+    assert response.status_code == 200
+    assert extract_value(response_dict, "FIRST_NAME") == user.first_name
+    assert extract_value(response_dict, "LAST_NAME") == user.last_name
+    assert extract_value(response_dict, "EMAIL") == user.email
+    assert extract_value(response_dict, "DATE_JOINED") == format_date(
+        user.date_joined  # type: ignore
```

Comment:
Isn't this a [mypy related ignore](https://mypy.readthedocs.io/en/stable/common_issues.html#spurious-errors-and-locally-silencing-the-checker)? Are you running the project through mypy? I haven't used mypy myself and I haven't seen it used much in the projects I've worked on so far. Does it work in the yjdh repository currently? Doesn't it give a lot of warnings if run? It might be interesting to include it in a project but as I have no experience with it I don't know what are the hurdles one has to go to through to really get a project work with it without any errors or warnings. We could have a talk about this as I am interested to hear about your experience with mypy if you have worked with it?